hooks/useClickOutside.ts

```diff
@@ -0,0 +1,21 @@
+import { useEffect, RefObject } from "react";
+
+interface UseClickOutsideProps {
+  ref: RefObject<HTMLElement>;
+  handler: () => void;
+}
+
+export function useClickOutside({ ref, handler }: UseClickOutsideProps) {
```

Comment:
p3: you can define ref in this hook

see: 

const ref = useClickOutside(() => { /* */})